apps/admin/backend/src/cvr_files.ts

```diff
@@ -394,3 +406,242 @@ export function convertCastVoteRecordToLegacyFormat({
     ...convertCastVoteRecordVotesToLegacyVotes(currentSnapshot),
   };
 }
+
+/**
+ *
+ */
+export type AddCastVoteRecordReportError =
+  | {
+      type: 'invalid-report-structure';
+      error: CastVoteRecordReportDirectoryStructureValidationError;
+    }
+  | {
+      type: 'malformed-report-metadata';
+      error: z.ZodError;
+    }
+  | {
+      type: 'malformed-cast-vote-record';
+      index: number;
+      error: z.ZodError;
+    }
+  | {
+      type: 'invalid-cast-vote-record';
+      index: number;
+      error: CastVoteRecordValidationError;
+    }
+  | {
+      type: 'invalid-report-file-mode';
+      currentFileMode: Admin.CvrFileMode;
+    }
+  | {
+      type: 'ballot-id-already-exists-with-different-data';
+    };
+
+/**
+ *
+ */
+export type AddCastVoteRecordReportResult = Result<
+  Admin.CvrFileImportInfo,
+  AddCastVoteRecordReportError
+>;
+
+/**
+ *
+ * @param param0
+ */
+export async function addCastVoteRecordReport({
+  store,
+  reportDirectoryPath,
+  exportedTimestamp,
+}: {
+  store: Store;
+  reportDirectoryPath: string;
+  exportedTimestamp: Iso8601Timestamp;
+}): Promise<AddCastVoteRecordReportResult> {
+  const electionId = store.getCurrentElectionId();
+  if (!electionId) {
+    throw new Error('no current election id');
+  }
```

Comment:
Nit: I prefer `assert` in these cases to express that we're expecting this to never happen